### ISSUE TITLE
Fix S6966 FP: EntityFrameworks DbContext/DBSet Add/AddRange methods are preferred over their Asnyc counterpart 

### ISSUE THREAD

**Issue by CaringDev** [SonarSource/sonar-dotnet#9269]
### Description

S6966 should not fire for DbContext.Add as AddAsync [is meant for special purposes only](https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbcontext.addasync?view=efcore-8.0): 
> This method is async only to allow special value generators, such as the one used by 'Microsoft.EntityFrameworkCore.Metadata.SqlServerValueGenerationStrategy.SequenceHiLo', to access the database asynchronously. For all other cases the non async method should be used.

### Repro steps

```csharp
public async Task Test()
{
    await using Microsoft.EntityFrameworkCore.DbContext ctx = new(null!);
    ctx.Add(null!); // S6966
}
```

### Expected behavior

No S6966 for DbContext.Add

### Actual behavior

S6966 is generated for the given example

### Known workarounds

Suppress warning (in code, config, ...)

### Related information

SonarAnalyzer.CSharp 9.25.0.90414

**Comment by martin-strecker-sonarsource** [comment]
Thank you, @CaringDev, for pointing this out. It is indeed a false positive. PR #9272 documents the misbehavior, and we will track this issue in our backlog.

**Comment by costin-zaharia-sonarsource** [comment]
Duplicated by #9301

**Comment by grosch-intl** [comment]
Using version 9.25.1.91650 this is still an issue.

![image](https://github.com/SonarSource/sonar-dotnet/assets/85708964/f3c9dc39-49a7-43b2-a754-1a3053789110)

**Comment by cremor** [comment]
I assume this will be fixed in 9.26? Then it should be assigned to [that milestone](https://github.com/SonarSource/sonar-dotnet/milestone/143).
